### PR TITLE
[Bug] fix the test_hgt function in test_nn.py

### PR DESCRIPTION
**Pull request by kunmukh**
## Description
The issue is with the mini-batch code. The block's  `num_dst_nodes` can sometimes be batch_size - 1, but the feature size is always batch_size. The test failed when the num_dst_nodes and feature sizes differed. II fixed it by adding a conditional to check if the size of the num_dst_nodes is the same as the batch size first.

## Checklist
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

## Changes
[x] separate input and output node types and handle them correctly in hgtconv.py
```
    if block.number_of_dst_nodes() == batch_size:
        y = m(block, x, ntype, etype)
```

### CODE REVIEW

File: tests/pytorch/test_nn.py [hunk 1]
```diff
@@ -1519,16 +1520,17 @@ def test_hgt(idtype, in_size, num_heads):
...
     (input_nodes, output_nodes, block) = next(iter(train_loader))
     block = block[0]
     x = x[input_nodes.to(th.long)]
     ntype = ntype[input_nodes.to(th.long)]
     edge = block.edata[dgl.EID]
     etype = etype[edge.to(th.long)]
-    y = m(block, x, ntype, etype)
-    assert y.shape == (block.number_of_dst_nodes(), head_size * num_heads)
+    if block.number_of_dst_nodes() == batch_size:
+        y = m(block, x, ntype, etype)
```
Comment (yaox12) [inline tests/pytorch/test_nn.py:1532]: But it means the sampled block cannot be used in training sometimes, right? If so, I believe there is a bug in the sampling algorithm.

Comment (mufeili) [reply tests/pytorch/test_nn.py:1532]: @BarclayII Any thoughts?

Comment (mufeili) [reply tests/pytorch/test_nn.py:1532]: @BarclayII Any thoughts?

Comment (BarclayII) [reply tests/pytorch/test_nn.py:1532]: I don't think this is the correct fix.  See #4947.

**Comment by BarclayII** [comment]
Closing this PR since the fix is not correct.  The problem lies in how the unit test generates the indices, not in how the HGTConv module handles the input.  See #4947 for more details.

**Pull request closed without merging**